compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/PropertiesToFieldsLowering.kt

```diff
@@ -72,8 +76,29 @@ class PropertiesToFieldsLowering(val context: CommonBackendContext) : IrElementT
         return accessor.origin == IrDeclarationOrigin.DEFAULT_PROPERTY_ACCESSOR && Visibilities.isPrivate(accessor.visibility)
     }
 
+    private val remappedFieldCache = mutableMapOf<IrField, IrField>()
+
+    private fun IrField.getRealBackingField(): IrField {
+        val irClass = resolveFakeOverride()!!.parent as? IrClass ?: return this
+        if ((irClass.isCompanion || (irClass.isObject && irClass.visibility != Visibilities.LOCAL)) && !isStatic) {
```

Comment:
Why is the visibility check necessary? I don't think local objects are possible. (Anonymous objects should have class kind `CLASS`.)

Reply:
I simply copied the condition from MoveOrCopyCompanionObjectFieldsLowering. You're right, it's redundant.

Reply:
Simlified in both places in the follow-up commit